src-java/testing/functional-tests/src/main/groovy/org/openkilda/functionaltests/helpers/model/HaFlowExtended.groovy

```diff
@@ -92,4 +106,78 @@ class HaFlowExtended {
                                boolean useTraffgenPorts = true, List<Integer> busyEndpoints = []) {
         return new HaFlowBuilder(swT, northboundV2, topologyDefinition, useTraffgenPorts, busyEndpoints)
     }
+
+    HaFlowUpdatePayload convertToUpdateRequest() {
+
+        String diverseLink = diverseWithFlows.isEmpty() ?
+                (diverseWithYFlows.isEmpty() ? (diverseWithHaFlows.isEmpty() ? null : diverseWithHaFlows[0])
+                        : diverseWithYFlows[0])
+                : diverseWithFlows[0]
+
+        def haFlowCopy = this.jacksonCopy()
+        def builder = HaFlowUpdatePayload.builder()
+        HaFlowUpdatePayload.class.getDeclaredFields()*.name.each {
+            builder.diverseFlowId(diverseLink)
+            if (haFlowCopy.class.declaredFields*.name.contains(it)) {
+                builder."$it" = haFlowCopy."$it"
+            }
+        }
+        return builder.build()
+    }
+
+    HaFlowHistory getHistory(Long timeFrom = null, Long timeTo = null, Integer maxCount = null) {
+        List<HaFlowHistoryEntry> historyRecords =  northboundV2.getHaFlowHistory(haFlowId, timeFrom, timeTo, maxCount)
+        return new HaFlowHistory (historyRecords)
+    }
+
+    boolean waitForDesiredFlowState(FlowState flowState) {
+        String flowCurrentStatus
+        Wrappers.wait(WAIT_OFFSET) {
+            flowCurrentStatus = northboundV2.getHaFlow(haFlowId).status
+            flowCurrentStatus == flowState.toString()
```

Comment:
It would be nice deserialize `status` field of response into FlowState object instead of String.

Reply:
Done.